### ISSUE TITLE
Custom prior and Discrete latents.

### ISSUE THREAD

**Issue by amasawa** [isjakewong/InfoDiffusion#1]
Hi yingheng, thanks for sharing the code, I have two questions regarding 'Custom prior and Discrete latents.' 

First, regarding the Custom prior mentioned in the article, does it mean that different datasets correspond to different priors? If so, could you please specify which prior was used for the dataset tested in the article? Also, when choosing different priors (Gaussian, Gaussian mixture, Swiss roll), we should be able to sample directly from these different priors, as stated in your Appendix D. However, this is not reflected in the code, where it seems to directly use Normal distribution for sampling.

Second, concerning Discrete latent, Appendix G describes how to handle discrete hidden features, but I didn't find any mention of Gumbel Softmax in the code.

Lastly, would it be possible for you to publish the hyperparameter settings for several datasets, including both training and testing (FID and DCI)?

**Comment by isjakewong** [comment]
Hi, thanks for your attention in our work. We test custom priors in our specific application scenarios but didn't include them in this paper. As you mentioned, we only use Gaussian thoughtout our experiments. We discussed this thoroughly in our another paper, you can check it out if you're especially interested in this: https://arxiv.org/html/2401.02739v1. Note that this paper is a bit outdated, we already updated the paper a lot and are preparing to release our second version very soon. Stay tuned! For the hyperparameters, I reported most of them in the appendix of the paper. But if you want to seek for a directly runnable version, I just uploaded a dev branch, which is the working version closest to the submission date. You can try running it. Training scripts and FID eval have already been included in this codebase I believe.

In terms of the discrete latent and DCI, they're in another branch of my co-author's codebase. I'll let them add these on soon.

**Comment by NafiseHaghshenas** [comment]
Could you kindly guide me on where I might find the code for Discrete Latent Priors used in your experiment? If it hasn't been included in the repository yet, would it be possible for you to share it? 
Thanks in advance for your help
Nafise
